openassessment/workflow/models.py

```diff
@@ -346,12 +355,59 @@ def _get_steps(self):
             # If no steps exist for this AssessmentWorkflow, assume
             # peer -> self for backwards compatibility
             self.steps.add(
-                AssessmentWorkflowStep(name=self.STATUS.peer, order_num=0),
-                AssessmentWorkflowStep(name=self.STATUS.self, order_num=1)
+                AssessmentWorkflowStep(name=self.STATUS.peer, order_num=1),
+                AssessmentWorkflowStep(name=self.STATUS.self, order_num=2)
             )
             steps = list(self.steps.all())
+        try:
+            self.steps.get(name=self.STATUS.staff)
+        except AssessmentWorkflowStep.DoesNotExist:
+            self.steps.add(
+                AssessmentWorkflowStep(
+                    name=self.STATUS.staff,
+                    order_num=0,
+                    assessment_completed_at=now(),
+                )
+            )
         return steps
 
+    def set_staff_score(self, score, is_override=False, reason=None):
+        """
+        Set a staff score for the workflow.
+
+        Allows for staff scores to bet set on a submission, with annotations to provide an audit trail if needed.
+        This method can be used for both required staff grading, and staff overrides.
+
+        Args:
+            score (dict): A dict containing 'points_earned', 'points_possible', and 'staff_id'.
+            is_override (bool): Optionally True if staff is overriding a previous score.
+            reason (string): An optional parameter specifying the reason for the staff grade. A default value
+                will be used in the event that this parameter is not provided.
+
+        """
+        annotation_type = STAFF_REQUIRED_ANNOTATION_TYPE
+        if is_override:
+            annotation_type = STAFF_OVERRIDE_ANNOTATION_TYPE
+            if reason is None:
+                reason = "A staff member has overridden a previous score for this submission"
+            sub_dict = sub_api.get_submission_and_student(self.submission_uuid)
+            sub_api.reset_score(
+                sub_dict['student_item']['student_id'],
+                self.course_id,
+                self.item_id
+            )
+        else:
+            if reason is None:
+                reason = "A staff score was required for this submission"
+        sub_api.set_score(
+            self.submission_uuid,
+            score["points_earned"],
+            score["points_possible"],
+            annotation_creator = score["staff_id"],
+            annotation_type = annotation_type,
+            annotation_reason = reason
+        )
```

Comment:
Do we need to worry about transactionality here? Theoretically a peer or other type of assessment could be recorded between `reset_score` and `set_score`. So
* does that matter, or will the fact that `staff` comes first in `get_score` priority save us anyways?
* how would we go about fixing this is if does matter? I'd think we'd need a new submissions api method that would allow for "reset and set" to happen in a single database transaction.